### ISSUE TITLE
Zombie bio-operator AI sometimes refuses to walk into melee

### ISSUE THREAD

**Issue by brndd** [CleverRaven/Cataclysm-DDA#38131]
# Describe the bug

Zombie bio-operators will sometimes not walk into melee, and will instead walk back and forth about 4-5 tiles away from the player for seemingly forever. This seems to be somehow related to aiming guns at them, as following the reproduction steps below while passing turns (with `5`) instead of aiming will cause the bio-operator to close into melee just fine.

# Steps To Reproduce

1. Load the [attached savegame](https://github.com/CleverRaven/Cataclysm-DDA/files/4219385/Parkersburg.zip).
2. Aim (with `f`ire) the shotgun at the bio-operator.
3. Steady your aim (with `.`) repeatedly.
4. The zombie will walk back and forth 4-5 tiles away from you instead of walking into melee.

# Expected behavior

For the zombie to walk into melee and kick my ass.

# Versions and configuration

- OS: Windows
    - OS Version: 10.0 1903
- Game Version: 0.D-12303-g790301c [64-bit]
- Graphics Version: Tiles
- Mods loaded: [
    Dark Days Ahead [dda],
    Disable NPC Needs [no_npc_food],
    No Fungal Monsters [No_Fungi],
    Prevent Zombie Revivication Redux [no_reviving_zombies_redux]
]

[Parkersburg.zip](https://github.com/CleverRaven/Cataclysm-DDA/files/4219385/Parkersburg.zip)

**Comment by KorGgenT** [comment]
i know zombies have the stumble() function, and i want confirmation that it's unintended behavior by looking through the code

**Comment by kevingranade** [comment]
This is very likely related to #38730

**Comment by brndd** [comment]
The save file attached should have experimental Z-levels disabled, in case that changes anything.

The bug in this case was triggered by aiming the gun at the zombie. If I loaded the attached save and then simply waited (with `.`) without aiming, the zombie would walk into melee without fail. Perhaps there is something in the AI that triggers when getting aimed at?

I can consistently reproduce this with the attached save and the specified game version (I have not updated my game in a while, so I haven't tested the latest experimental version). I noticed now that the save requires a custom mod, but the required mod simply disables zombie revives and blacklists zombie necromancers, so it should be safe to simply remove it as a dependency.
